mypyc/genops.py

```diff
@@ -3996,15 +3998,23 @@ def emit_yield(self, val: Value, line: int) -> Value:
         assert cls.send_arg_reg is not None
         return cls.send_arg_reg
 
-    def visit_yield_from_expr(self, o: YieldFromExpr) -> Value:
+    def handle_yield_from_and_await(self, o: Union[YieldFromExpr, AwaitExpr]) -> Value:
         # This is basically an implementation of the code in PEP 380.
 
         # TODO: do we want to use the right types here?
         result = self.alloc_temp(object_rprimitive)
         to_yield_reg = self.alloc_temp(object_rprimitive)
         received_reg = self.alloc_temp(object_rprimitive)
-        iter_reg = self.maybe_spill_assignable(
-            self.primitive_op(iter_op, [self.accept(o.expr)], o.line))
+
+        if isinstance(o, YieldFromExpr):
+            iter_val = self.primitive_op(iter_op, [self.accept(o.expr)], o.line)
+        else:
+            # If the object is a coroutine, call its' await method to get the iterator
```

Comment:
Like we discussed offline, the `await` case needs to also try `iter` if `__await__` isn't there, and we want to directly do the access through the pointer tables instead of with a getattr

Reply:
This is done now